python/llama-index-server/llama_index/server/services/workflow.py

```diff
@@ -0,0 +1,122 @@
+import json
+import logging
+from pathlib import Path
+from typing import Type
+
+from llama_index.core.workflow import (
+    Context,
+    Event,
+    JsonSerializer,
+    Workflow,
+)
+from llama_index.core.workflow.handler import WorkflowHandler
+from llama_index.server.utils.class_meta_serialization import (
+    type_from_identifier,
+    type_identifier,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class HITLWorkflowService:
+    """
+    A service for helping pause and resume a HITL workflow.
+    """
+
+    # A key in context that stores the HITL event type
+    HITL_CONTEXT_KEY = "human_response_type"
+
+    @staticmethod
+    def get_storage_path(id: str) -> Path:
+        storage_dir = Path("output") / "checkpoints"
+        if not storage_dir.exists():
+            storage_dir.mkdir(parents=True, exist_ok=True)
+        return storage_dir / f"{id}.json"
+
+    @classmethod
+    async def save_context(
+        cls,
+        id: str,
+        ctx: Context,
+        resume_event_type: Type[Event],
```

Comment:
as it's now a HITL service, maybe:
```suggestion
        resume_event_type: Type[HumanInputEvent],
```

Reply:
Updated, but it's better to use `resume_event_type: Type[HumanResponseEvent]`